### PR TITLE
fix: improve twitter post content quality

### PR DESCRIPTION
**Pull request by cygaar**
# Relates to:

Issues with duplicate twitter content being posted

# Risks

Low

# Background

## What does this PR do?

Reduces duplication of Twitter content and reduces tokens used for twitter post generation.

The two key changes were:
1) Removing previous posts from the generation prompt as this influenced the model into repeating previous content even though the prompt asked it not to. Previous posts do not provide value when generation posts. For tweet interactions we already include exists posts in the thread so including previous posts (timeline) is not necessary.
2) Improve the prompts to remove unused blank lines and to better organize the structure of the context. These changes should improve response quality.

## What kind of change is this?

Improvement

# Documentation changes needed?

# Testing

### CODE REVIEW

File: packages/client-twitter/src/interactions.ts
Comment (cygaar) [inline packages/client-twitter/src/interactions.ts:232]: This is incorrectly named "timeline" - it was really just a list of the bot's previous posts

Comment (odilitime) [reply packages/client-twitter/src/interactions.ts:232]: Meh, I’ve heard it called timeline. Profile timeline might be more apt and precise

**Comment by twilwa** [comment]
LGTM, as long as that discussion didn't change a variable name and break anything

**Comment by odilitime** [comment]
Not a review just comments:
While it’s hard to argue with results. I think the context of the recent public tweets could be useful for keep the agent focused and less random. But that can be reintroduced later when it’s more thought through and tested. 

I also feel like some comments in the code explaining these changes would be helpful, so they’re not reverted.

**Review by twilwa — APPROVED** [review]
LGTM

File: packages/client-twitter/src/interactions.ts
Comment (cygaar) [reply packages/client-twitter/src/interactions.ts:232]: yeah, we can revisit this in a future pr. didn't actually change the underlying function

**Comment by cygaar** [comment]
I think since we're already providing a fair amount of context to the model about the agent (knowledge, topics), the responses tend to not be very random. They all seem relatively related to eachother even without providing the last few tweets.

File: packages/client-twitter/src/interactions.ts
Comment (fabianhug) [reply packages/client-twitter/src/interactions.ts:232]: lgtm, if we need to have a list of previous posts for some analysis, we can quickly re-introduce it. For the PR its good

File: packages/client-twitter/src/post.ts
Comment (fabianhug) [inline packages/client-twitter/src/post.ts:31]: a bit cleaner would be something like this. Maybe we can fine-tune or A/B test the prompt. 

`Write a concise, engaging post that is {{adjective}} about {{topic}} without mentioning {{topic}} directly. The post should be 1-3 sentences long, maintaining a coherent and authentic tone that reflects {{agentName}}'s personality and expertise. Avoid adding any commentary or acknowledgment of this request. Ensure the total character count is less than 280 characters. Use \\n\\n (double spaces) between statements. No emojis or questions allowed.`

**Review by fabianhug — APPROVED** [review]
lgtm, I added a comment regarding the prompt.

@lalalune mentioned this PR in his Discord stream, but I tuned in right when he stopped. Let's wait for his comment before merging

File: packages/client-twitter/src/post.ts
Comment (cygaar) [reply packages/client-twitter/src/post.ts:31]: I can try this out

**Pull request merged**
